ui/SpinInfoBar.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import { useDark } from './lib/useDark';
+
+export const SpinInfoBar = () => {
+  const [spinStack, setSpinStack] = useState<string[]>([]);
+  const dark = useDark();
+
+  useEffect(() => {
+    console.log('spin: setting up...');
+    let available = false;
+    const spin = new EventSource('/~_~/spin', { withCredentials: true });
+
+    spin.onopen = () => {
+      console.log('spin: opened stream');
+      available = true;
+    };
+
+    spin.onmessage = (e) => {
+      // The data is expected to be a path like /stack3/stack2/stack1/stack0
+      // We need to parse it into an array of stack names
+      if (e.data && typeof e.data === 'string') {
+        // Remove leading slash if present and split by slash
+        const stackPath = e.data.substring(1);
+        const stackArray = stackPath.split('/');
+        setSpinStack(stackArray);
+      }
+    };
+
+    spin.onerror = (e) => {
+      console.error('spin: eventsource error:', e);
+    };
+
+    return () => {
+      spin.close();
+    };
+  }, []);
+
+  if (spinStack.length === 0) {
+    return null;
+  }
```

Comment:
This causes it to not render until we've heard something from the spin endpoint. Should be nearly instant usually, but you can imagine cases (problematic cases) where that takes a hot second or longer. This causes a rough "pop-in" effect, especially given that styling was updated so that the terminal window doesn't hit the bottom of the viewport until this element comes into existence.

Should just render it unconditionally.